File: src/main/java/toker/warbandscripts/panel/entity/Profession.java
package toker.warbandscripts.panel.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import org.springframework.cache.annotation.Cacheable;

import javax.persistence.*;
import java.util.Collection;
import java.util.Objects;

@Entity
@Table(name = "profession")
public class Profession {
    private Integer id;
    private String name;
    private Integer max_tier;
    private Collection<ProfessionAssignment> professionAssignments;
    private Integer maxTier;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    @Basic
    @Column(name = "name", nullable = false, length = 32)
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Basic
    @Column(name = "max_tier", nullable = false)
    public Integer getMaxTier() {
        return max_tier;
    }

    public void setMaxTier(Integer max_tier) {
        this.max_tier = max_tier;
    }

    @OneToMany(mappedBy = "profession")
    @JsonIgnore
    public Collection<ProfessionAssignment> getProfessionAssignments() {
        return professionAssignments;
    }

    public void setProfessionAssignments(Collection<ProfessionAssignment> professionAssignments) {
        this.professionAssignments = professionAssignments;
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Profession profession = (Profession) o;
        return Objects.equals(id, profession.id);
    }
}

File: src/main/java/toker/panel/entity/Profession.kt
package toker.panel.entity

import com.fasterxml.jackson.annotation.JsonIgnore
import javax.persistence.*

@Entity
@Table(name = "profession")
class Profession(
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @Column(name = "id", nullable = false)
        var id: Int? = null,

        @Column(name = "name", nullable = false, length = 32)
        var name: String? = null,
) {
    @OneToMany(mappedBy = "profession")
    @JsonIgnore
    var professionAssignments: MutableSet<ProfessionAssignment>? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        other as Profession
        if (id != other.id) return false
        return true
    }

    override fun hashCode(): Int {
        return id ?: 0
    }
}